gtda/diagrams/features.py

```diff
@@ -522,6 +522,172 @@ def transform(self, X, y=None):
+    separately and regarded as sums of Dirac deltas.
+    Then, the convolution with a Gaussian kernel is computed over
+    a rectangular grid of locations evenly sampled from appropriate ranges
+    of the `filtration parameter <https://giotto.ai/theory>`_.
+    The result can be thought of as a raster image.
+
+    Parameters
+    ----------
+    sigma : float, optional default ``1.0``
+        Standard deviation for Gaussian kernel.
+
+    n_bins : int, optional, default: ``100``
+        The number of filtration parameter values, per available homology
+        dimension, to sample during :meth:`fit`.
+
+    weight_function : fct 1d array -> ad array, default: p -> p
+        Function mapping a 1d-array of persistence of the points of a diagram
+        to a 1d array of their weight.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    homology_dimensions_ : list
+        Homology dimensions seen in :meth:`fit`.
+
+    samplings_ : dict
+        For each number in `homology_dimensions_`, a discrete sampling of
+        filtration parameters, calculated during :meth:`fit` according to the
+        minimum birth and maximum death values observed across all samples.
+
+    weights_ : dict
+        For each number in `homology_dimensions_`, an array of weights
+        corresponding to the persistence values obtained from `samplings_`
+        calculated during :meth:`fit` using the `weight_function`.
+
+    See also
+    --------
+    BettiCurve, PersistenceLandscape, PersistenceEntropy, HeatKernel, \
+    Amplitude, PairwiseDistance, gtda.homology.VietorisRipsPersistence
+
+    Notes
+    -----
+    The samplings in :attr:`samplings_` are in general different between
+    different homology dimensions. This means that the (i, j)-th pixel of a
+    persistence image in homology dimension q typically arises from a different
+    pair of parameter values to the (i, j)-th pixel of a persistence image in
+    dimension q'.
+
+    References
+    ----------
+    .. [1] H. Adams, T. Emerson, M. Kirby, R. Neville, C. Peterson, P. Shipman,
+           S. Chepushtanova, E. Hanson, F. Motta, and L. Ziegelmeier,
+           "Persistence Images: A Stable Vector Representation of Persistent
+           Homology"; *Journal of Machine Learning Research 18, 1*,
+           pp. 218-252, 2017; doi: `10.5555/3122009.3122017
+           <http://dx.doi.org/10.5555/3122009.3122017>`_.
+
+    """
+
+    _hyperparameters = {'sigma': [numbers.Number, (1e-16, np.inf)],
+                        'n_bins': [int, (1, np.inf)],
+                        'weight_function': [types.FunctionType]}
+
+    def __init__(self, sigma=1.0, n_bins=100,
+                 weight_function=lambda x: x,
```

Comment:
Should i keep the default value in the docstring though?
```weight_function : fct 1d array -> ad array, default: p -> p```
My reasoning is that it is more informative than the 'actual' default `None`, which, I would say, is the result of some technical issues (cf your comment).

Reply:
Also, it seems like I would have to exclude this parameter from `_hyperparameters`, since `validate_params`can take only one type. (or, refactor the check)

Reply:
Could you not check the "effective" version of the parameter, as it's done elsewhere?

Reply:
I have done it.